core/trino-spi/src/main/java/io/trino/spi/connector/MetadataProvider.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import java.util.Optional;
+
+public interface MetadataProvider
```

Comment:
I find the name very generic. I'd be comfortable with a grainer scoped name fitting to the purpose of this interface.  Same thing for `InternalMetadataProvider`

NOTE that there exists already `io.trino.metadata.Metadata` and it is not related to this interface

Reply:
What would you suggest?